utils/utils.go

```diff
@@ -172,3 +173,32 @@ func GetUrlWithCache(url string, cache *RequestCache, skip bool) ([]byte, error)
 	cache.Date = resp.Header.Get("Date")
 	return cache.Data, err
 }
+
+// Append cookies to passed baseUrl into query params
+func BuildUrlQueryStringFromCookies(baseUrl string, r *http.Request) string {
+	u, err := url.Parse(baseUrl)
+	if err != nil {
+		return ""
+	}
+	q, err := url.ParseQuery(u.RawQuery)
+	if err != nil {
+		return ""
+	}
+	for _, cookie := range r.Cookies() {
+		q.Add(cookie.Name, cookie.Value)
+	}
+	u.RawQuery = q.Encode()
+	return u.String()
+}
+
+var DefaultUrlSchemes = []string{"http", "https", "ftp", "mailto", "tel"}
+
+// Validate .. if passed redirectUrl is a valid mobile app custom scheme url
+// Scheme should not match any default schemes
+func IsValidCustomSchemeUrl(customSchemeUrl string) bool {
```

Comment:
Shouldn't this just be looking for the prefix `mattermost://`? Won't, say, `file://foo` or `facetime://bar` incorrectly appear valid?

Reply:
We have ["Build your own app"](https://developers.mattermost.com/contribute/mobile/build-your-own/) and they would have different scheme url. That's why I thought of making it a part of config on the mobile client.

If needed we can also think about making it a config like "Site url" ??

Reply:
Added it in the config properties.